From 67a9d3c4f56643d59274d5cf491e089ab5f8d43b Mon Sep 17 00:00:00 2001
From: Ricardo Salveti de Araujo <ricardo.salveti@canonical.com>
Date: Fri, 24 Jan 2014 21:19:45 -0200
Subject: [PATCH 05/10] Acodec: No need for death notifier

With Ubuntu Touch we don't use the media service for video decoding.

Change-Id: I52fc09eaf837b2c16571dbb40f43f8a569a200f7
Signed-off-by: Ricardo Salveti de Araujo <ricardo.salveti@canonical.com>
Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 media/libstagefright/ACodec.cpp    | 9 +++++----
 media/libstagefright/OMXClient.cpp | 2 +-
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 05f8131..8213d37 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -5237,12 +5237,13 @@ bool ACodec::UninitializedState::onAllocateComponent(const sp<AMessage> &msg) {
 
     sp<AMessage> notify = new AMessage(kWhatOMXDied, mCodec->id());
 
-    mDeathNotifier = new DeathNotifier(notify);
-    if (omx->asBinder()->linkToDeath(mDeathNotifier) != OK) {
+    // Media service is not used with Ubuntu Touch
+    // mDeathNotifier = new DeathNotifier(notify);
+    // if (omx->asBinder()->linkToDeath(mDeathNotifier) != OK) {
         // This was a local binder, if it dies so do we, we won't care
         // about any notifications in the afterlife.
-        mDeathNotifier.clear();
-    }
+        // mDeathNotifier.clear();
+    // }
 
     Vector<OMXCodec::CodecNameAndQuirks> matchingCodecs;
 
diff --git a/media/libstagefright/OMXClient.cpp b/media/libstagefright/OMXClient.cpp
index a109f02..87c6690 100644
--- a/media/libstagefright/OMXClient.cpp
+++ b/media/libstagefright/OMXClient.cpp
@@ -37,7 +37,7 @@ struct MuxOMX : public IOMX {
     MuxOMX(const sp<IOMX> &remoteOMX);
     virtual ~MuxOMX();
 
-    virtual IBinder *onAsBinder() { return mRemoteOMX->asBinder().get(); }
+    virtual IBinder *onAsBinder() { return NULL; }
 
     virtual bool livesLocally(node_id node, pid_t pid);
 
-- 
2.7.4

